Neos.Diff/Classes/SequenceMatcher.php

```diff
@@ -0,0 +1,723 @@
+     */
+    private function isBJunk($b)
+    {
+        if (isset($this->juncDict[$b])) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Find the longest matching block in the two sequences, as defined by the
+     * lower and upper constraints for each sequence. (for the first sequence,
+     * $alo - $ahi and for the second sequence, $blo - $bhi)
+     *
+     * Essentially, of all of the maximal matching blocks, return the one that
+     * starts earliest in $a, and all of those maximal matching blocks that
+     * start earliest in $a, return the one that starts earliest in $b.
+     *
+     * If the junk callback is defined, do the above but with the restriction
+     * that the junk element appears in the block. Extend it as far as possible
+     * by matching only junk elements in both $a and $b.
+     *
+     * @param int $alo The lower constraint for the first sequence.
+     * @param int $ahi The upper constraint for the first sequence.
+     * @param int $blo The lower constraint for the second sequence.
+     * @param int $bhi The upper constraint for the second sequence.
+     * @return array Array containing the longest match that includes the starting position in $a, start in $b and the length/size.
+     */
+    public function findLongestMatch($alo, $ahi, $blo, $bhi)
+    {
+        $a = $this->a;
+        $b = $this->b;
+
+        $bestI = $alo;
+        $bestJ = $blo;
+        $bestSize = 0;
+
+        $j2Len = [];
+        $nothing = [];
+
+        for ($i = $alo; $i < $ahi; ++$i) {
+            $newJ2Len = [];
+            $jDict = $this->arrayGetDefault($this->b2j, $a[$i], $nothing);
+            foreach ($jDict as $jKey => $j) {
+                if ($j < $blo) {
+                    continue;
+                } else {
+                    if ($j >= $bhi) {
+                        break;
+                    }
+                }
+
+                $k = $this->arrayGetDefault($j2Len, $j - 1, 0) + 1;
+                $newJ2Len[$j] = $k;
+                if ($k > $bestSize) {
+                    $bestI = $i - $k + 1;
+                    $bestJ = $j - $k + 1;
+                    $bestSize = $k;
+                }
+            }
+
+            $j2Len = $newJ2Len;
+        }
+
+        while ($bestI > $alo && $bestJ > $blo && !$this->isBJunk($b[$bestJ - 1]) &&
+            !$this->linesAreDifferent($bestI - 1, $bestJ - 1)) {
+            --$bestI;
+            --$bestJ;
+            ++$bestSize;
+        }
+
+        while ($bestI + $bestSize < $ahi && ($bestJ + $bestSize) < $bhi &&
+            !$this->isBJunk($b[$bestJ + $bestSize]) && !$this->linesAreDifferent($bestI + $bestSize,
+                $bestJ + $bestSize)) {
+            ++$bestSize;
+        }
+
+        while ($bestI > $alo && $bestJ > $blo && $this->isBJunk($b[$bestJ - 1]) &&
+            !$this->isLineDifferent($bestI - 1, $bestJ - 1)) {
```

Comment:
should probably also be `linesAreDifferent`, because `isLineDifferent` doesn't exist…